### PR TITLE
Add ultrametricize and is_ultrametric functions for phylogenies

### PR DESCRIPTION
**Pull request by mmore500**
## Summary
This PR adds functionality to adjust phylogeny tip times to be ultrametric (all tips at the same time) and to check if a phylogeny is already ultrametric. Both Pandas and Polars implementations are provided with comprehensive test coverage.

## Key Changes

- **New Pandas functions**:
  - `alifestd_ultrametricize()`: Adjusts tip `origin_time` values so all tips share the same time using the "extend" method (sets all tips to the maximum tip time)
  - `alifestd_is_ultrametric()`: Checks if all tips share the same `origin_time` within a tolerance

- **New Polars functions**:
  - `alifestd_ultrametricize_polars()`: High-performance Polars implementation of ultrametricization
  - `alifestd_is_ultrametric_polars()`: Polars implementation of ultrametric checking

- **CLI entrypoints**:
  - `phyloframe.legacy._alifestd_ultrametricize`: Pandas-based CLI with support for CSV and Parquet formats
  - `phyloframe.legacy._alifestd_ultrametricize_polars`: Polars-based CLI for high-performance processing

- **Comprehensive test suites**:
  - 291 lines of Pandas tests covering edge cases (empty phylogenies, forests, sexual reproduction, non-contiguous IDs, etc.)
  - 253 lines of Polars tests with both DataFrame and LazyFrame variants
  - 244 lines of is_ultrametric tests for Pandas
  - 246 lines of is_ultrametric tests for Polars
  - CLI integration tests for both implementations

## Implementation Details

- Both implementations respect pre-existing `is_leaf` columns if present, allowing for custom leaf definitions
- The "extend" method sets all leaf nodes to the maximum `origin_time` among leaves while leaving internal nodes unchanged
- Polars implementation delegates to `alifestd_mark_leaves_polars` for leaf detection when needed
- Pandas implementation includes optional in-place mutation via the `mutate` parameter
- `is_ultrametric` functions support an `atol` parameter for floating-point tolerance comparisons
- Input dataframes are not mutated unless explicitly requested (Pandas only)
- Both implementations handle empty phylogenies, forests with multiple roots, and sexual phylogenies
- Updated `__init__.pyi` and `__main__.py` to export new functions and document CLI entrypoints

https://claude.ai/code/session_01XLp256y8KFzcnEHSxyoV8W

### CODE REVIEW

File: phyloframe/legacy/_alifestd_is_ultrametric_polars.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    if "is_leaf" not in schema_names:
+        phylogeny_df = alifestd_mark_leaves_polars(phylogeny_df)
+
+    leaf_stats = (
+        phylogeny_df.lazy()
+        .filter(pl.col("is_leaf"))
+        .select(
+            pl.col("origin_time").max() - pl.col("origin_time").min(),
+            pl.col("origin_time").is_null().any().alias("any_null"),
```
Comment (mmore500) [inline phyloframe/legacy/_alifestd_is_ultrametric_polars.py]: aliases are unnecessary

File: phyloframe/legacy/_alifestd_ultrametricize.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            f"alifestd_ultrametricize: unknown method {method!r}",
+        )
+
+    if not mutate:
+        phylogeny_df = phylogeny_df.copy()
+
+    if phylogeny_df.empty:
+        return phylogeny_df
+
+    latest_origin_time = phylogeny_df["origin_time"].max()
```
Comment (mmore500) [inline phyloframe/legacy/_alifestd_ultrametricize.py]:
```suggestion
    latest_origin_time = phylogeny_df["origin_time"].max(skipna=True)
```

File: phyloframe/legacy/_alifestd_is_ultrametric.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import numpy as np
+import pandas as pd
+
+from ._alifestd_mark_leaves import alifestd_mark_leaves
+
+
+def alifestd_is_ultrametric(
+    phylogeny_df: pd.DataFrame,
+    *,
+    atol: float = 0.0,
```
Comment (mmore500) [inline phyloframe/legacy/_alifestd_is_ultrametric.py:11]: need mutate kwarg (look for other examples of this in pandas funcs)

**Review by mmore500 — COMMENTED** [review]
lgtm, ready to merge after CI and comments addressed
